src/Http/Routing/src/ValidationEndpointFilterFactory.cs

```diff
@@ -25,12 +40,10 @@ public static EndpointFilterDelegate Create(EndpointFilterFactoryContext context
 
         var serviceProviderIsService = context.ApplicationServices.GetService<IServiceProviderIsService>();
 
-        var parameterCount = parameters.Length;
-        var validatableParameters = new IValidatableInfo[parameterCount];
-        var parameterDisplayNames = new string[parameterCount];
-        var hasValidatableParameters = false;
+        // Use a list to only store validatable parameters instead of arrays for all parameters
+        var validatableParameters = new System.Collections.Generic.List<ValidatableParameterEntry>();
```

Comment:
Instead of using fully-qualified type names, add a using for `System.Collections.Generic` and use `new List<ValidatableParameterEntry>()`.